ql/src/test/results/clientpositive/llap/compare_cols_null.q.out

```diff
@@ -34,8 +34,9 @@ STAGE PLANS:
       Processor Tree:
         TableScan
           alias: ccn_table
+          filterExpr: (key > '123a') (type: boolean)
           Filter Operator
-            predicate: false (type: boolean)
+            predicate: (key > '123a') (type: boolean)
```

Comment:
A little question: would we do a full table scan when the predicate is  ` (key > '123a')` here, or cloud we also do something optimization for these cases?

Reply:
@dengzhhu653 , that's a great point. The path moving forward should be to fold/optimize this type of expressions in the optimizer (Calcite) rather than in `TypeCheckProcFactory.java`. It seems we have a gap in the optimization logic, that's why it's not getting folded. @scarlin-cloudera , could you create a follow-up JIRA to fix this?